src/teehr/loading/nwm/nwm_grids.py

```diff
@@ -0,0 +1,195 @@
+from typing import Union, Iterable, Optional
+from datetime import datetime
+from pathlib import Path
+
+from teehr.loading.nwm.grid_utils import fetch_and_format_nwm_grids
+from teehr.loading.nwm.utils import (
+    build_remote_nwm_filelist,
+    generate_json_paths,
+    check_dates_against_nwm_version
+)
+from teehr.models.loading.utils import (
+    SupportedNWMOperationalVersionsEnum,
+    SupportedNWMDataSourcesEnum,
+    SupportedKerchunkMethod
+)
+from teehr.loading.nwm.const import (
+    NWM22_UNIT_LOOKUP,
+    NWM22_ANALYSIS_CONFIG,
+    NWM30_ANALYSIS_CONFIG,
+)
+
+
+def nwm_grids_to_parquet(
+    configuration: str,
+    output_type: str,
+    variable_name: str,
+    start_date: Union[str, datetime],
+    ingest_days: int,
+    zonal_weights_filepath: str,
+    json_dir: Union[str, Path],
+    output_parquet_dir: Union[str, Path],
+    nwm_version: SupportedNWMOperationalVersionsEnum,
+    data_source: Optional[SupportedNWMDataSourcesEnum] = "GCS",
+    kerchunk_method: Optional[SupportedKerchunkMethod] = "create",
+    t_minus_hours: Optional[Iterable[int]] = None,
+    ignore_missing_file: Optional[bool] = True,
+    overwrite_output: Optional[bool] = False,
+):
+    """
+    Fetches NWM gridded data, calculates zonal statistics (mean) of selected
+    variable for given zones, converts and saves to TEEHR tabular format
+
+    Parameters
+    ----------
+    configuration : str
+        NWM forecast category.
+        (e.g., "analysis_assim", "short_range", ...)
+    output_type : str
+        Output component of the configuration.
+        (e.g., "channel_rt", "reservoir", ...)
+    variable_name : str
+        Name of the NWM data variable to download.
+        (e.g., "streamflow", "velocity", ...)
+    start_date : str or datetime
+        Date to begin data ingest.
+        Str formats can include YYYY-MM-DD or MM/DD/YYYY
+    ingest_days : int
+        Number of days to ingest data after start date
+    zonal_weights_filepath: str
+        Path to the array containing fraction of pixel overlap
+        for each zone
+    json_dir : str
+        Directory path for saving json reference files
+    output_parquet_dir : str
+        Path to the directory for the final parquet files
+    nwm_version: SupportedNWMOperationalVersionsEnum
+        The NWM operational version
+        "nwm22", or "nwm30"
+    data_source: Optional[SupportedNWMDataSourcesEnum]
+        Specifies the remote location from which to fetch the data
+        "GCS" (default), "NOMADS", or "DSTOR"
+        Currently only "GCS" is implemented
+    kerchunk_method: Optional[SupportedKerchunkMethod]
+        When data_source = "GCS", specifies the preference in creating Kerchunk
+        reference json files.
+        "create" - (default) always create new json files from netcdf files in GCS and
```

Comment:
@samlamont when using the "create" method, if it exists locally will it still create it?  Or is this the same as the old behavior where it will look for a local file?

Reply:
The old behavior is retained, so existing local files will still be used (not created) if they exists

Reply:
we should probably indicate that in the doc string - "always create" is misleading. I wonder if maybe some different key words would be more intuitive. Maybe "local", "remote", "auto", and then make sure the behavior is clear.

Reply:
sure sounds good agreed.  I'll update the variable names and docs